src/renderer/widgets/WalletDetails/ui/WalletConnectDetails.tsx

```diff
@@ -0,0 +1,87 @@
+import { useMemo } from 'react';
+import { useUnit } from 'effector-react';
+
+import { Wallet, Chain, Account, WalletConnectAccount } from '@renderer/shared/core';
+import { BaseModal, BodyText, StatusLabel } from '@renderer/shared/ui';
+import { DEFAULT_TRANSITION } from '@renderer/shared/lib/utils';
+import { useToggle } from '@renderer/shared/lib/hooks';
+import { MultiAccountsList, WalletIcon } from '@renderer/entities/wallet';
+import { useI18n } from '@renderer/app/providers';
+import { chainsService } from '@renderer/entities/network';
+import { walletProviderModel } from '../model/wallet-provider-model';
+
+type AccountItem = {
+  accountId: `0x${string}`;
+  chain: Chain;
+};
+
+type Props = {
+  isOpen: boolean;
+  wallet: Wallet;
+  accounts: Account[];
+  onClose: () => void;
+};
+export const WalletConnectDetails = ({ isOpen, wallet, accounts, onClose }: Props) => {
+  const { t } = useI18n();
+
+  const [isModalOpen, toggleIsModalOpen] = useToggle(isOpen);
+
+  const connected = useUnit(walletProviderModel.$connected);
+
+  const closeWowModal = () => {
+    toggleIsModalOpen();
+
+    setTimeout(onClose, DEFAULT_TRANSITION);
+  };
+
+  const accountsList = useMemo(() => {
+    const chains = chainsService.getChainsData();
+    const sortedChains = chainsService.sortChains(chains);
+
+    const accountsList = sortedChains.reduce<AccountItem[]>((acc, c) => {
+      const account = accounts.find((a) => c.chainId === (a as WalletConnectAccount).chainId);
```

Comment:
Not that important right now, but if have time and come up with a solution that is not N*N, that would be awesome 🤩
Maybe create a map of `accounts/chainIds`